### PR TITLE
Clean up and singular value decomposition

### PR DESCRIPTION
**Pull request by utkuozbulak**
Made a bit of clean up and formatting, and Singular Value Decomposition :smiling_imp: 
Breakdown:

Functions related to data are removed from data2img to format_data (csv2img)
Removed main from data2img
Added few more functions to data2img.py to save single image, zoom image etc.
Added main, ( main main ) and structure is more similar to what we have in data mining.
Added a function that returns the data in numpy matrix format which we can use in conv nets directly
Added svd.py which has many functions related to Singular Value Decomposition, comments are there
Added an example decomposition in the main at the bottom, its super cool :laughing:  check it ! :heart_eyes: 
Everything runs on main now with a single file read ( except for Alex's files, I only updated mine and Lukas' )

### CODE REVIEW

File: src/format_data.py [hunk 1]
```diff
@@ -1,28 +1,74 @@
...
 
-if __name__ == "__main__":
-    raw_data = pd.read_csv(raw_data_csv_file_name)
+
+def pandas_vector_to_list(pandas_df):
+    py_list = [item[0] for item in pandas_df.values.tolist()]
+    return py_list
+
+
+def get_data_in_vectoral_form(emotion, pixels):
```
Comment (utkuozbulak) [inline src/format_data.py]: Part of clean up, already reviewed before

File: src/format_data.py [hunk 2]
```diff
@@ -1,28 +1,74 @@
...
+    y_data_categorical = np_utils.to_categorical(y_data, 7)
+    # Y data
+    y_train = np.array(y_data_categorical[0:TRAIN_END])
+    y_public_test = np.array(y_data_categorical[PUBLIC_TEST_START:PUBLIC_TEST_END])
+    y_private_test = np.array(y_data_categorical[PRIVATE_TEST_START:])
+    return (x_train_vectoral, x_public_test_vectoral, x_private_test_vectoral,
+            y_train, y_public_test, y_private_test)  # Returns data in vectoral format
+
+
+def csv2array(emotion, pixels, img_size=IMG_SIZE):
```
Comment (utkuozbulak) [inline src/format_data.py:41]: Part of clean up, already reviewed before

File: src/format_data.py [hunk 3]
```diff
@@ -1,28 +1,74 @@
...
+        data = np.zeros((img_size, img_size), dtype=np.uint8)
+        pixel_data = item.split()
+        for i in range(0, img_size):
+            pixel_index = i * img_size
+            data[i] = pixel_data[pixel_index:pixel_index + img_size]
+        img_list.append((emotion_as_list[index], data))
+    return img_list  # Returns a tuple, emotion:pixel array
+
+
+def get_data_in_matrix_format(emotion, pixels):
```
Comment (utkuozbulak) [inline src/format_data.py:55]: A new function that uses csv2array and returns it in matrix format with train/test sets ( it was tuple in csv2array )

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    # BEWARE ! The line below takes incredibly long to run because it is calculating a decomposition
+    # For each image, hence its commented out for now
+    # decomposed_image_list = generate_decomposed_matrices_from_list(x_train_matrix)
+    # An example decomposition
+    example_image = x_train_matrix[4]  # Example image from matrix form, e.g: fourth image
+    U,S,V = decompose(example_image)  # Decomposition
+    decomposed_list = generate_decomposed_matrices(U, S, V)  # Decomposed list, 48 matrices
+    # You can play with the number '10' and observe different results
+    # As the sum of decompositions increase the summed image gets better and better
+    save_sample_decomposition(decomposed_list, 10)  # Saves some sample results of this decomposition
```
Comment (utkuozbulak) [inline src/main.py:39]: 10 is the amount of decomposed matrices that will be used ( total of 48 there )

File: src/svd.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from data2img import save_single_img, get_zoomed_image
+
+
+def decompose(image_matrix):
+    """
+    # Get decomposed matrices, USVtranspose
+    """
+    U, S, V = np.linalg.svd(image_matrix, full_matrices=True)
```
Comment (alexdy2007) [inline src/svd.py:9]: Very little syntax thing, no need for full_matrices=True

def svd(a, full_matrices=1, compute_uv=1):
    """
    Singular Value Decomposition.

Default is full_matrices = True as in python:     1 == True :-)

File: src/data2img.py [hunk 6]
```diff
@@ -40,20 +14,32 @@ def display_img(data):
...
+
+def show_zoomed_image(image_as_list ,zoom_ratio):
+    resized_list = imresize(image_as_list, zoom_ratio)
+    resized_img = toimage(resized_list)
+    resized_img.show()
+    return resized_img
+
+
+def get_zoomed_image(image_as_list ,zoom_ratio):
+    resized_list = imresize(image_as_list, zoom_ratio)
```
Comment (alexdy2007) [inline src/data2img.py:37]: may produce sharper image if set imresize(interp="bicubic"). Maths is a lost on me but wikipedia says so 

'In image processing, bicubic interpolation is often chosen over bilinear interpolation or nearest neighbor in image resampling, when speed is not an issue.' 
https://en.wikipedia.org/wiki/Bicubic_interpolation

**Review by alexdy2007 — COMMENTED** [review]
Nice work, only few very minor syntax stuff.

**Review by alexdy2007 — APPROVED** [review]
I'll approve, assume everyone else ok with this?

**Pull request merged**